### ISSUE TITLE
Cannot build on Visual Studio 17.7.3

### ISSUE THREAD

**Issue by mdblack98** [ramdor/Thetis#150]
Get errors on C# version incompatibility.
.Net  4.5.1 only supports V7.3 and one need  V8 in order to use  nullable fields.
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS8630	Invalid 'nullable' value: 'Enable' for C# 7.3. Please use language version '8.0' or greater.	Midi2Cat	C:\Users\mdbla\Dropbox\Projects\OpenHPSDR-Thetis\Project Files\Source\Midi2Cat\CSC	1	Active
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS8630	Invalid 'nullable' value: 'Enable' for C# 7.3. Please use language version '8.0' or greater.	RawInput	C:\Users\mdbla\Dropbox\Projects\OpenHPSDR-Thetis\Project Files\Source\RawInput\CSC	1	Active
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS8630	Invalid 'nullable' value: 'Enable' for C# 7.3. Please use language version '8.0' or greater.	Thetis	C:\Users\mdbla\Dropbox\Projects\OpenHPSDR-Thetis\Project Files\Source\Console\CSC	1	Active

**Comment by laurencebarker** [comment]
I’m using version 17.1.1. It does seem that we all need to move forward in sync. I don’t know what other developers have installed. 

 

 

Laurence Barker G8NJJ

***@***.***

 

From: Michael Black ***@***.***> 
Sent: Tuesday, September 5, 2023 5:48 AM
To: ramdor/Thetis ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ramdor/Thetis] Cannot build on Visual Studio 17.7.3 (Issue #150)

 

Get errors on C# version incompatibility.
.Net 4.5.1 only supports V7.3 and one need V8 in order to use nullable fields.
Severity Code Description Project File Line Suppression State
Error CS8630 Invalid 'nullable' value: 'Enable' for C# 7.3. Please use language version '8.0' or greater. Midi2Cat C:\Users\mdbla\Dropbox\Projects\OpenHPSDR-Thetis\Project Files\Source\Midi2Cat\CSC 1 Active
Severity Code Description Project File Line Suppression State
Error CS8630 Invalid 'nullable' value: 'Enable' for C# 7.3. Please use language version '8.0' or greater. RawInput C:\Users\mdbla\Dropbox\Projects\OpenHPSDR-Thetis\Project Files\Source\RawInput\CSC 1 Active
Severity Code Description Project File Line Suppression State
Error CS8630 Invalid 'nullable' value: 'Enable' for C# 7.3. Please use language version '8.0' or greater. Thetis C:\Users\mdbla\Dropbox\Projects\OpenHPSDR-Thetis\Project Files\Source\Console\CSC 1 Active

**Comment by DH1KLM** [comment]
I'm not a developer, but would like to report that I can't find any errors when compiling on 17.7.4.
The latest update for NET Framework 3.5, 4.8 and 4.8.1 for Windows 10 version 22H2 is installed.

**Comment by mdblack98** [comment]
I upgraded to 17.7.4 but that didn't fix it.
I had to change all these (there were several on each build entry) from disable to enable in Thetis.csproj and RawInput.csproj
<Nullable>enable</Nullable>

Now it works.
Mike W9MDB 

    On Wednesday, September 13, 2023 at 09:11:49 AM CDT, Sigi ***@***.***> wrote:  
 
 


I'm not a developer, but would like to report that I can't find any errors when compiling on 17.7.4.
The latest update for NET Framework 3.5, 4.8 and 4.8.1 for Windows 10 version 22H2 is installed.
